lib/market.rb

```diff
@@ -0,0 +1,60 @@
+require 'pry'
+class Market
+  attr_reader :name, :vendors
+  def initialize(name, vendors = [])
+    @name = name
+    @vendors = vendors
+    @names = []
+  end 
+
+  def add_vendor(vendor)
+    @vendors << vendor
+  end
+
+  def vendor_names
+    vendor_names_array = [] #created a new array for the vendor names
+    @vendors.each do |vendor|
+      vendor_names_array << vendor.name
+    end
+    vendor_names_array
+  end
+
+  def vendors_that_sell(item)
+    vendors_that_sell = []  # create an empty array
+    @vendors.each do |vendor|  # Iterate through each vendor in the vendors array
+      # Check if the vendor has stock for the specified item.
+      if vendor.check_stock(item) > 0
+        vendors_that_sell << vendor  # If the vendor has the item in stock, add the vendor to the vendors_that_sell
+      end
+    end
+    vendors_that_sell  # Returns the array
+  end
+
+  def total_inventory
```

Comment:
Great job on this method!  One thing to note from your video submission - try to use `block variable` instead of 'label' to describe what is in the pipes after `each do`